config/deploy.rb

```diff
@@ -44,3 +44,79 @@
     end
   end
 end
+
+namespace :container do
+  desc 'Hello, world!'
+  task :hello do
+    on roles(:container) do
+      execute "echo Hello, world!"
+    end
+  end
+
+  desc 'deploy'
+  task :deploy do
+    on roles(:container) do
+      execute "sudo docker pull ryuzee/slidehub:latest"
+      prefix = DateTime.now.strftime('%Y%m%d%H%M%s')
+      cmd =<<"EOS"
+        docker run -d \
+        --env OSS_REGION=#{fetch(:oss_region)} \
+        --env OSS_SQS_URL=#{fetch(:oss_sqs_url)} \
+        --env OSS_BUCKET_NAME=#{fetch(:oss_bucket_name)} \
+        --env OSS_IMAGE_BUCKET_NAME=#{fetch(:oss_image_bucket_name)} \
+        --env OSS_USE_S3_STATIC_HOSTING=#{fetch(:oss_use_s3_static_hosting)} \
+        --env OSS_AWS_SECRET_KEY=#{fetch(:oss_aws_secret_key)} \
+        --env OSS_AWS_ACCESS_ID=#{fetch(:oss_aws_access_id)} \
+        --env OSS_SECRET_KEY_BASE=#{fetch(:oss_secret_key_base)} \
+        --env OSS_DB_NAME=#{fetch(:oss_db_name)} \
+        --env OSS_DB_USERNAME=#{fetch(:oss_db_username)} \
+        --env OSS_DB_PASSWORD=#{fetch(:oss_db_password)} \
+        --env OSS_DB_URL=#{fetch(:oss_db_url)} \
+        -P --name slidehub#{prefix} ryuzee/slidehub
+EOS
+      container_id = capture("sudo #{cmd}")
+      port = capture("sudo docker port #{container_id} 3000").to_s.split(':')[1]
+      puts port
+      # confirm running
+      cmd = "curl -LI http://127.0.0.1:#{port} -o /dev/null -w '%{http_code}\\n' -s"
+      cnt = 0
+      while true do
+        execute "echo 'sleep 20 sec...'"
+        sleep 20
+        cnt += 1
+        if cnt == 10
+          error = Exception.new("An error that should abort and rollback deployment")
```

Comment:
__[rubocop]__
* Style/StringLiterals: Prefer single-quoted strings when you don't need string interpolation or special symbols. [:ref](https://github.com/bbatsov/ruby-style-guide#consistent-string-literals)